Duplicati/Library/Common/IO/SystemIOWindows.cs

```diff
@@ -407,14 +539,7 @@ public void FileCopy(string source, string target, bool overwrite)
 
         public string PathGetFullPath(string path)
         {
-            if (IsPrefixedWithUNC(path))
-            {
-                return System.IO.Path.GetFullPath(ConvertSlashes(path));
-            }
-            else
-            {
-                return StripUNCPrefix(System.IO.Path.GetFullPath(PrefixWithUNC(ConvertSlashes(path))));
-            }
+            return PrefixWithUNC(Path.GetFullPath(path));

```

Comment:
I think that's probably wrong.  Let me try to get it straight in my head.  The cases I'd like to be handled are:

1. If path is already prefixed with `\\?\`, it should be left completely alone since, by definition, it should not require any additional "parsing".
2. If path is not already prefixed with `\\?\`, the return value also should not be prefixed
3. If path is relative or has relative components, that should be resolved by calling Path.GetFullPath()
4. If path is not relative and has no relative components, I don't want Path.GetFullPath() to process it as-is because it will munge the "problematic Windows paths"

Maybe the following works:
```
        public string PathGetFullPath(string path)
        {
            if (IsPrefixedWithUNC(path))
            {
                return path;
            }
            else
            {
                return StripUNCPrefix(Path.GetFullPath(PrefixWithUNC(path)));
            }
        }
```
For my cases above, case 1 is handled by the first leg of the if statement.  Case 2 is handled by the call to StripUNCPrefix.  Case 3 works because PrefixWithUNC() will leave path unprefixed and will pass it, as-is, to Path.GetFullPath().  Case 4 works because PrefixWithUNC() will prefix with `\\?\`, thereby protecting "problematic Windows paths" from Path.GetFullPath(), then StripUNCPrefix strips the protection.

Reply:
I've just pushed a commit to address this.